include/boost/typeof/typeof.hpp

```diff
@@ -15,7 +15,7 @@
 
 #include <boost/config.hpp>
 
-#if !defined(BOOST_NO_CXX11_DECLTYPE) && !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) && !defined(BOOST_TYPEOF_EMULATION)
+#if (!defined(_MSC_VER) || (_MSC_VER > 1900)) && !defined(BOOST_NO_CXX11_DECLTYPE) && !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) && !defined(BOOST_TYPEOF_EMULATION)
```

Comment:
Since this fix is specific to MSVC, it should test for `BOOST_MSVC`.

Reply:
Done